Portal/vip-core/src/main/java/fr/insalyon/creatis/vip/core/server/dao/mysql/PlatformConnection.java

```diff
@@ -59,154 +59,164 @@
+                            + "PRIMARY KEY (email, groupname), "
+                            + "FOREIGN KEY (email) REFERENCES VIPUsers(email) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE, "
+                            + "FOREIGN KEY (groupname) REFERENCES VIPGroups(groupname) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE");
+
+                createTable(connection,
+                            "VIPAccounts",
+                            "name VARCHAR(255), "
+                            + "PRIMARY KEY (name)");
+
+                createTable(connection,
+                            "VIPAccountsGroups",
+                            "name VARCHAR(255), "
+                            + "groupname VARCHAR(255), "
+                            + "PRIMARY KEY (name, groupname), "
+                            + "FOREIGN KEY (name) REFERENCES VIPAccounts(name) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE, "
+                            + "FOREIGN KEY (groupname) REFERENCES VIPGroups(groupname) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE");
+
+                createTable(connection,
+                            "VIPPublication",
+                            "id INT(11) NOT NULL AUTO_INCREMENT, "
+                            + "title VARCHAR(255) NULL, "
+                            + "date VARCHAR(45) NULL, "
+                            + "doi VARCHAR(255) NULL, "
+                            + "autors VARCHAR(255) NULL, "
+                            + "type VARCHAR(255) NULL, "
+                            + "typeName VARCHAR(255) NULL, "
+                            + "VIPAuthor VARCHAR(255) NULL, "
+                            + "PRIMARY KEY (id), "
+                            + "FOREIGN KEY (VIPAuthor) REFERENCES VIPUsers(email) "
+                            + "ON DELETE CASCADE ON UPDATE CASCADE");
+
+                createTable(connection,
+                            "VIPTermsOfuse",
+                            "idTermsOfuse INT(11) NOT NULL AUTO_INCREMENT, "
+                            + "date TIMESTAMP NULL, "
+                            + "text TEXT NULL, "
+                            + "PRIMARY KEY (idTermsOfuse)");
+
+                firstExecution = false;
+            } catch (SQLException e) {
+                logger.error("Error closing database connection", e);
+            }
         }
     }
 
-    public Connection getConnection() {
+    // We throw a SQLException and not a DAOException because the client of this
+    // function must anyway catch the SQLException when closing the connection.
+    // This avoids him to have to catch 2 types of exceptions.
+    public Connection getConnection() throws SQLException {
+        Connection connection = dataSource.getConnection();
+        Connection actual = ((javax.sql.PooledConnection)connection).getConnection();
+        logger.info("actual connection : " + actual.toString());
```

Comment:
I"m not sure to fully understand the aim of the "actual" Connection. Is it only for logging purposes or is it more (the doc says that "The connection pool manager calls this method when an application has called the method DataSource.getConnection and there are no PooledConnection objects available.") ?

Reply:
Yes, getting the actual connection was done only for logging purpose.  I'll remove the log, it's too verbose.